clients/src/main/java/org/apache/kafka/clients/consumer/internals/NetworkClientDelegate.java

```diff
@@ -254,28 +250,38 @@ public String toString() {
         }
     }
 
-    public static class FutureCompletionHandler implements RequestCompletionHandler {
+    public static class FutureCompletionHandler extends CompletableFuture<ClientResponse> implements RequestCompletionHandler {
 
-        private final CompletableFuture<ClientResponse> future;
+        /**
+         * The time when the response is completed. This is used when the response is completed exceptionally because
+         * ClientResponse already contains received time which is injected by the network client.
+         */
```

Comment:
Thanks, I think I was trying to say the main use of responseCompletionTimeMs is mainly used when the future gets completed exceptionally.  For the successful requests, we could use either the ClientResponse or this time.  I'll remove this comment to avoid the ambiguity.